blog/crsp-v2-update/index.qmd

```diff
@@ -0,0 +1,133 @@
+
+::: {.panel-tabset group="language"}
+## R
+
+```{r}
+#| eval: false
+msf_db <- tbl(wrds, in_schema("crsp", "msf_v2"))
+stksecurityinfohist_db <- tbl(wrds, in_schema("crsp", "stksecurityinfohist"))
+```
+
+## Python
+
+```{python, eval = FALSE, python.reticulate = FALSE}
+#| eval: false
+crsp_monthly_query = (
+  "SELECT msf.permno"
+    "FROM crsp.msf_v2 AS msf "
+    "LEFT JOIN crsp.stksecurityinfohist AS ssih "
+  )
+```
+:::
+
+We do not repost the new routines for downloading the data in this blog. Instead, we ask you to check out the new code in the respective chapter on **WRDS, CRSP, and Compustat** in the [R version](../../r/wrds-crsp-and-compustat.qmd) and the [Python version](../../python/wrds-crsp-and-compustat.qmd).
+
+## A small glimpse at the differences
+
+To take a look at the new table and compare it to the old, we use the new `tidyfinance` R package to download the data. If you do not have the package yet, please install it from CRAN using `install.packages("tidyfinance")`. For ease of use, we also load the full `tidyverse` as in all our chapters of the R book.
+
+```{r}
+#| message: false
+#| warning: false
+library(tidyfinance)
+library(tidyverse)
+```
+
+Then, we can download the two data formats with the function `download_data_wrds_crsp`.
+
+```{r}
+#| warning: false
+#| message: false
+
+crsp_old <- download_data_wrds_crsp("crsp_monthly", 
+                                    version = "v1",
+                                    start_date = "1970-01-01",
+                                    end_date = "2022-12-01")
+
+crsp_new <- download_data_wrds_crsp("crsp_monthly", 
+                                    version = "v2",
+                                    start_date = "1970-01-01",
+                                    end_date = "2022-12-01")
+```
+
+Now, do we find any significant differences between the data? Let us look at the number of observations identified by `permno` and `date`.
+
+```{r}
+nrow(crsp_new)
+nrow(crsp_old)
+
+anti_join(crsp_new, crsp_old,
+          by = join_by("permno", "date")) |> 
+  nrow()
+```
+We find that roughly 4,000 observations are not matched, where the new data contains more rows. We are not sure where these observations come from or where we lost them in the old data. However, this is a very minor difference. 
+
+Next, we check if the monthly returns are all the same. In fact, they should not be as CRSP states that they have updated the way to compound distributions. They also state this change in compounding as an example among other things. However, they do not tell us about these *other* things.
+
+```{r}
+inner_join(crsp_new |> 
+             select(permno, date, ret_new = ret), 
+           crsp_old |> 
+             select(permno, date, ret_old = ret),
+          by = join_by("permno", "date")) |> 
+  mutate(differences = near(ret_new, ret_old, tol = 10^-6)) |> 
+  summarize(differences = all(differences, na.rm = TRUE))
```

Comment:
Wouldn't it be more informative to have the number of rows with differences? 

mutate(ret_differences = !near(ret_new, ret_old, tol = 10^-6)) |>summarize(share_ret_differences = sum(ret_differences) / n())